gax-java/gax/src/main/java/com/google/api/gax/rpc/EndpointContext.java

```diff
@@ -100,6 +102,11 @@ public static EndpointContext getDefaultInstance() {
   @Nullable
   public abstract String transportChannelProviderEndpoint();
 
+  public abstract boolean useS2A();
+
+  @Nullable
+  public abstract EnvironmentProvider envProvider();
```

Comment:
Does `envProvider` needs to be public? Is seems only used within `shouldUseS2A()`

Reply:
It does not need to public, done in https://github.com/googleapis/sdk-platform-java/pull/3326/commits/943683db2e9996dc1d767bb5a8cd26b532fba3e2